### PR TITLE
feature: HTTP status codes

### LINKED ISSUE #1977: Idea: Add Http status codes
* Faker version: 22.2.0
* OS: all

Feature suggestion : add http status code to `providers.internet` as a random integer between 100 and 599

source: https://www.rfc-editor.org/rfc/rfc9110#name-status-codes

### PR DESCRIPTION
**Pull request by dancergraham**
### What does this change

Adds HTTP status codes - closes #1977 

### What was wrong

All Http status codes must be in the range 100 to 599, with a subset of these values being currently assigned by RFCs

### How this fixes it

Selects by default a random value in the full valid range or optionally a random selection from only the assigned values

Fixes #1977

### CODE REVIEW

**Comment by dancergraham** [comment]
ouch - segfault in pypy - not sure why ?

File: tests/providers/test_internet.py [hunk 1]
```diff
@@ -339,6 +340,12 @@ def test_http_method(self, faker, num_samples):
 
         assert expected_methods == sorted(got_methods)
 
+    def test_http_status_code(self, faker, num_samples):
+        provider = InternetProvider(faker)
+        status_code = provider.http_status_code()
+        assert isinstance(status_code, int)
+        assert 100 <= status_code <= 599
+
```
Comment (fcurella) [inline tests/providers/test_internet.py:348]: can we also have a test for `include_unassigned=False`?

**Review by fcurella — CHANGES_REQUESTED** [review]

**Comment by fcurella** [comment]
> ouch - segfault in pypy - not sure why ?

yeah, that's been happening for. while and I'm not sure why, it's been erratic 🤷🏼 

I've re-run the tests and the build is green now

File: tests/providers/test_internet.py (hunk 1 above)
Comment (dancergraham) [reply tests/providers/test_internet.py:351]: OK - done!

**Review by fcurella — APPROVED** [review]
Thank you!

**Pull request merged**